src/main/java/hudson/plugins/timestamper/pipeline/GlobalAnnotator.java

```diff
@@ -99,19 +88,16 @@
         return this;
     }
 
-    /**
-     * Parse this line for a timestamp starting at position {@code start}, if such a timestamp is
-     * present.
-     */
+    /** Parse this line for a timestamp if such a timestamp is present. */
     @Restricted(NoExternalUse.class)
-    public static Optional<Timestamp> parseTimestamp(String html, int start, long buildStartTime) {
-        if (html.startsWith("[", start)) {
-            int end = html.indexOf(']', start);
+    public static Optional<Timestamp> parseTimestamp(String text, Run<?, ?> build) {
+        if (text.startsWith("[")) {
+            int end = text.indexOf(']');
             if (end != -1) {
                 try {
-                    long millisSinceEpoch = ZonedDateTime.parse(html.substring(start + 1, end), GlobalDecorator.UTC_MILLIS).toInstant().toEpochMilli();
-                    // Alternately: Instant.parse(html.substring(start + 1, end)).toEpochMilli()
-                    Timestamp timestamp = new Timestamp(millisSinceEpoch - buildStartTime, millisSinceEpoch);
+                    long millisSinceEpoch = ZonedDateTime.parse(text.substring(1, end), GlobalDecorator.UTC_MILLIS).toInstant().toEpochMilli();
+                    // Alternately: Instant.parse(text.substring(1, end)).toEpochMilli()
+                    Timestamp timestamp = new Timestamp(millisSinceEpoch - build.getStartTimeInMillis(), millisSinceEpoch);
```

Comment:
I would prefer the original `long buildStartTime` parameter as it is more amenable to unit testing.

Reply:
Good point. I'll restore the original `long` parameter.